diff --git a/Makefile b/Makefile
index b4175da..37c8747 100755
--- a/Makefile
+++ b/Makefile
@@ -29,7 +29,7 @@ test: $(MININET) $(TEST)
 
 mnexec: mnexec.c $(MN) mininet/net.py
 	$(CC) $(CFLAGS) $(LDFLAGS) \
-	-DVERSION=\"`PYTHONPATH=. XTABLES_LIBDIR=/usr/lib/x86_64-linux-gnu/xtables $(PYMN) --version 2>&1`\" $< -o $@
+	-DVERSION=\"`PYTHONPATH=. XTABLES_LIBDIR=/usr/lib/x86_64-linux-gnu/xtables $(PYMN) --version 2>&1 | grep -oE '[0-9a-z]+\.[0-9a-z]+\.[0-9a-z]+'`\" $< -o $@
 
 install-mnexec: $(MNEXEC)
 	install -D $(MNEXEC) $(BINDIR)/$(MNEXEC)
